### PR TITLE
Submodule history added to calling module history

### PR DESCRIPTION
**Pull request by dehall**
Modifies the CallSubmodule state so that the history of a submodule is added to the history of the calling module, as is currently done in the Ruby version. This should be preserved even if the submodule delays across multiple timesteps.

### CODE REVIEW

File: src/main/java/org/mitre/synthea/engine/Module.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
  */
 public class Module {
 
-  private static Map<String, Module> modules = Collections.unmodifiableMap(loadModules());
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/engine/Module.java]: Why did you change this? Making the collection unmodifiable was done to make it read only. Changing it to final will make the property read-only, but you'll now be able to edit the contents of the modules map. I don't think we want to allow this

Comment (dehall) [reply src/main/java/org/mitre/synthea/engine/Module.java]: I didn't really want to do this either, but because the CallSubmodule state gets the submodule to call from this collection of modules, any test modules have to be in there if we want to test the CallSubmodule state. I can't see any way to have the test modules be loaded automatically when testing and not when running the real thing. (and we don't necessarily want all the test modules loaded when testing anyway, some of them intentionally have errors)

There's almost certainly a "better" way to implement Module and CallSubmodule such that they can be tested with test fixtures but offhand I would assume it to be a large refactor.

I'm open to suggestions.

Comment (dehall) [reply src/main/java/org/mitre/synthea/engine/Module.java]: Following up on our offline discussion - the Module.getModules() method already creates a new list every time it's called. So because this field is private it can only be accessed from here (or via reflection) and there's no serious concern that another place might accidentally modify it.
